sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala

```diff
@@ -217,6 +217,27 @@ class UDFRegistration private[sql] (functionRegistry: FunctionRegistry) extends
     if (nullable) udf else udf.asNonNullable()
   }
 
+  /**
+   * Registers a deterministic Scala closure of 2 arguments as user-defined function (UDF).
+   * @since 2.4.0
+   */
+  def registerV2(name: String, func: Function2[_, _, _], returnType: DataType,
+    input1Type: DataType, input2Type: DataType, nullable: Boolean, deterministic: Boolean)
```

Comment:
`deterministic ` should be part of UDF definition, instead of a parameter of register function.

Reply:
so you mean:
```
def registerV2(udf: UserDefinedFunctionV2) 
```
?